### ISSUE TITLE
Can we download the generated QR?

### ISSUE THREAD

**Issue by howdyhyber** [ushelp/EasyQRCodeJS#10]
Greetings!

This could be a great feature if this will be added. :D

**Comment by cyoung502** [comment]
> Greetings!
> 
> This could be a great feature if this will be added. :D


Here is a simple solution.  Generate the qr-code and with a button add a click event to save the code, or a button that generates and saves, same thing.  EasyQRCodeJS generates the output as a png, so we look for the img element, get the src which is a DataURI.  Next we can just add that to an anchor tag attribute href and set the download attribute to the filename for all modern broswers.  However Internet Explorer doesn't support download attribute, so we must convert the DataURI to a blob and use the msSaveBlob() function in IE to prompt the user to save the image.
```
$("button").click(function (e) {
    function dataURItoBlob(dataURI) {
        for (var byteString = atob(dataURI.split(",")[1]), mimeString = dataURI.split(",")[0].split(":")[1].split(";")[0], ab = new ArrayBuffer(byteString.length), ia = new Uint8Array(ab), i = 0; i < byteString.length; i++)
            ia[i] = byteString.charCodeAt(i);
        var blob;
        return new Blob([ab], {
            type: mimeString
        })
    }
    var dataUri = $("#qrcode").children("img").attr("src");
    if (navigator.msSaveBlob) {
        var blob = dataURItoBlob(dataUri);
        return navigator.msSaveBlob(blob, url + "-qrcode.png")
    } else {
        $(this).attr("href", dataUri),
            $(this).attr("download", url + "-qrcode.png")
    }
});
```

**Comment by howdyhyber** [comment]
Do you have any react solution for this?

**Comment by ushelp** [comment]
@howdyhyber 
Good Job!
But I think if you need to download images, you can try to generate images on the server side. It can avoid the deficiency of many browser doms.
[https://github.com/ushelp/EasyQRCodeJS-NodeJS ](https://github.com/ushelp/EasyQRCodeJS-NodeJS )

**Comment by ushelp** [comment]
The download function has been released, please update your library to 4.5.0 or above.

Please refer to: 
https://github.com/ushelp/EasyQRCodeJS#download-the-qrcode-image-in-the-browser
